src/utils/components/AddBootableVolumeModal/utils/utils.ts

```diff
@@ -71,10 +87,87 @@ export const createBootableVolume =
       };
     });
 
-    const newDataSource = await k8sCreate({ data: dataSourceToCreate, model: DataSourceModel });
+    isUploadForm
+      ? await uploadData({
+          dataVolume: bootableVolumeToCreate,
+          file: uploadFile as File,
+        })
+      : await k8sCreate({ data: bootableVolumeToCreate, model: DataVolumeModel });
+
+    newDataSource = await k8sCreate({ data: dataSourceToCreate, model: DataSourceModel });
 
     onCreateVolume?.(newDataSource);
+
+    return newDataSource;
   };
 
 export const getInstanceTypeFromVolume = (bootableVolume: BootableVolume): string =>
   bootableVolume?.metadata?.labels?.[DEFAULT_INSTANCETYPE_LABEL] ?? '';
+
+type CreateDataSourceWithImportCronType = (
+  bootableVolume: AddBootableVolumeState,
+  initialDataSource: V1beta1DataSource,
+) => Promise<V1beta1DataSource>;
+
+export const createDataSourceWithImportCron: CreateDataSourceWithImportCronType = async (
+  bootableVolume,
+  initialDataSource,
+) => {
+  const { bootableVolumeName, cronExpression, registryURL, retainRevisions, size } = bootableVolume;
+
+  const dataImportCronName = `${bootableVolumeName}-import-cron-${getRandomChars()}`;
+  const dataImportCron = produce(initialDataImportCron, (draft) => {
+    draft.metadata.name = dataImportCronName;
+    draft.metadata.namespace = initialDataSource?.metadata?.namespace;
+    draft.spec = {
+      garbageCollect: 'Outdated',
+      importsToKeep: retainRevisions,
+      managedDataSource: bootableVolumeName,
+      schedule: cronExpression,
+      template: {
+        spec: {
+          source: {
+            registry: {
+              url: registryURL.includes('docker://') ? registryURL : 'docker://' + registryURL,
```

Comment:
i think we have functions of dockerPrefix or something like that, please check

Reply:
Okay. The function is called `appendDockerPrefix` and it's defined in the catalog. 
After this pr I'll create another pr to move this function in the utils folder.